### PR TITLE
Woche 04 aufgaben

### PR DESCRIPTION
**Pull request by TomKnoden**

### CODE REVIEW

File: Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+def bearbeite_programm(programm, eingabe):
+    position = 0
+    ausgabe = []
+    while True:
+        instruktion = programm[position]
+        opcode, modus1, modus2, modus3 = instruktion % 100, (instruktion // 100) % 10, (instruktion // 1000) % 10, (instruktion // 10000) % 10
+
+        if opcode == 1:  # Addition
+            param1 = programm[position + 1] if modus1 == 1 else programm[programm[position + 1]]
```
Comment (thecodingcube77) [inline Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py:9]: You could leave out the `== 1` as anything non zero is interpreted as `True`

File: Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        elif opcode == 99:  # Beenden
+            break
+        else:
+            raise ValueError("Ungültiger Opcode")
+
+    return ausgabe
+
+# Beispiel-Eingabe und Ausführung des Programms
+programm = [int(x) for x in input("Geben Sie das Programm als Liste von Zahlen ein: ").split(",")]
+eingabe = int(input("Geben Sie die Eingabe (z.B. 5) ein: ")
```
Comment (thecodingcube77) [inline Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py:57]: The input should be in the `bearbeite_programm()` function under opcode 3. This is problematic as there could possibly be more than one input.

File: Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            param2 = programm[position + 2] if modus2 == 1 else programm[programm[position + 2]]
+            programm[programm[position + 3]] = param1 + param2
+            position += 4
+        elif opcode == 2:  # Multiplikation
+            param1 = programm[position + 1] if modus1 == 1 else programm[programm[position + 1]]
+            param2 = programm[position + 2] if modus2 == 1 else programm[programm[position + 2]]
+            programm[programm[position + 3]] = param1 * param2
+            position += 4
+        elif opcode == 3:  # Eingabe
+            programm[programm[position + 1]] = eingabe
```
Comment (thecodingcube77) [inline Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py:19]: The input should happen here inside of the function.

File: Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            position += 2
+        elif opcode == 4:  # Ausgabe
+            ausgabe.append(programm[programm[position + 1]] if modus1 == 0 else programm[position + 1])
```
Comment (thecodingcube77) [inline Exercises/Tom_Knoden/Woche04/Woche_04_Aufgaben.py:22]: The value should be printed here and not added to an array.

**Review by thecodingcube77 — CHANGES_REQUESTED** [review]
First of all you did not state what your output was in your pull request. Please change that.

Otherwise the solution seems almost okay to me. You got really much duplicate code which you could try to remove in some way (that is not mainly what stops me from approving, just a general comment).

The problems can be seen in detail in the other comments.

**Comment by atrejut** [comment]
@TomKnoden please try to get this old PR merged

**Pull request closed without merging**